### PR TITLE
Prefer IO::NULL over hardcoded device name

### PR DESCRIPTION
**Pull request by nobu**

### CODE REVIEW

**Comment by vipulnsward** [comment]
Maybe all these can be extracted as a single method?

File: activerecord/test/cases/migration_test.rb [hunk 1]
```diff
@@ -939,7 +939,15 @@ def quietly
 
   def silence_stream(stream)
     old_stream = stream.dup
-    stream.reopen(RbConfig::CONFIG['host_os'] =~ /mswin|mingw/ ? 'NUL:' : '/dev/null')
+    null = case
+           when IO.const_defined?(:NULL)
```
Comment (egilburg) [inline activerecord/test/cases/migration_test.rb]: On which platforms is IO::NULL not defined?

Comment (nobu) [inline activerecord/test/cases/migration_test.rb]: 1.9.2 and earlier.

Comment (egilburg) [inline activerecord/test/cases/migration_test.rb]: 1.9.2 hasn't been supported by Rails for a long time. Even for existing 4.2 build, 1.9.3 is minimum.

As for Rails 5.0, it'll have minimum version of either 2.2 or 2.1.

So I think you could just always use IO:NULL and remove the other conditions

**Comment by senny** [comment]
@nobu I merged a rebased version of this PR. Thank you :yellow_heart:

**Comment by senny** [comment]
@vipulnsward I agree that it would be nice to extract a method into a common place for all test suites. As far as I know other than `active_support/testing` there is no shared code at the moment. Feel free to submit a PR to explore possible solutions.

**Pull request closed without merging**